src/pages/Provider/Provider.layout.vue

```diff
@@ -1,91 +1,40 @@
 <template>
   <div class="Provider">
     <provider-header
-      :logo-url="provider.logo_url"
-      :name="provider.commercial_name"
-      v-model:query-value="searchText"
-      @onSearch="getProductLists(id, searchText)"
-    />
-    <categories-slider
-      class="Provider__categories"
-      :categories="[]"
-      :categorySelected="''"
+      :logo-url="provider?.logo_url"
+      :name="provider?.commercial_name"
+      :query-value="searchText"
+      @update:query-value="$emit('update:searchText', $event)"
+      @onSearch="$emit('onSearch')"
     />
     <div class="Provider__container">
-      <suspense>
-        <template #default>
-          <provider-products :products="products" />
-        </template>
-        <template #fallback>
-          <base-loading />
-        </template>
-      </suspense>
+      <base-loading v-if="isLoading" />
+      <provider-products v-else :products="products" />
     </div>
   </div>
 </template>
 
-<script lang="ts">
-import { defineComponent, PropType, ref } from 'vue';
+<script setup lang="ts">
 import { BaseLoading } from '../../components/LoadingComponent';
-import { getProductsByProvider } from '../../actions';
-import providerHeader from './ProviderHeader.vue';
-import ProviderProducts from './ProviderProducts.vue';
-import CategoriesSlider from '../../components/CategoriesSlider';
-import { Product, provider } from 'src/utils';
-
-const handleProviderProducts = () => {
-  const products = ref<Product[]>([]);
-  const query = ref<string>('');
-  const searchText = ref<string>('');
+import { ProviderHeader, ProviderProducts } from './components';
+import './Provider.scss';
 
-  const getProductLists = async (id: string, query: string) => {
-    if (id) {
-      await getProductsByProvider(id, query).then((res) => {
-        products.value = res;
-      });
-    }
-  };
+import { Product, Provider } from 'src/utils';
 
-  return {
-    getProductLists,
-    query,
-    products,
-    searchText,
-  };
-};
+interface ProviderLayoutProps {
+  id: string;
+  provider?: Provider;
+  products: Product[];
+  searchText: string;
```

Comment:
Hmm is ok I think it can be nullable, but doesn't should be nullable since the developer should always provide the state, without that prop the search bar doesn't work properly.

But in case I will add some of protection since the prop could come as null / undefined